Sources/SwiftRefactor/ConvertZeroParameterFunctionToComputedProperty.swift

```diff
@@ -0,0 +1,69 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=6)
+public import SwiftSyntax
+#else
+import SwiftSyntax
+#endif
+
+public struct ConvertZeroParameterFunctionToComputedProperty: SyntaxRefactoringProvider {
+  public static func refactor(syntax: FunctionDeclSyntax, in context: ()) -> VariableDeclSyntax? {
+    guard syntax.signature.parameterClause.parameters.isEmpty,
+      let body = syntax.body
+    else { return nil }
+
+    let variableName = PatternSyntax(
+      IdentifierPatternSyntax(
+        leadingTrivia: syntax.funcKeyword.trailingTrivia,
+        identifier: syntax.name
+      )
+    )
+
+    let paramsTrailingTrivia = syntax.signature.parameterClause.rightParen.trailingTrivia
+    let colonLeadingTrivia = paramsTrailingTrivia == .space ? [] : paramsTrailingTrivia
```

Comment:
I think we shouldn’t be dropping any trivia from the parameter clause

```suggestion
let triviaFromParameters = (syntax.signature.parameterClause.leftParen.trivia + syntax.signature.parameterClause.rightParen.trivia).droppingTrailingWhitespace
```

With `trivia` and `droppingTrailingWhitespace` from your [other PR](https://github.com/swiftlang/swift-syntax/pull/2712/files#diff-5d23988c50101027622d00dbafeddfe2b73ba76ad27953908617600f8a8c2721R86-R96)

I think that requires a few test updates but I think all of them are improvements.